starter/src/main/java/com/termux/x11/starter/Starter.java

```diff
@@ -0,0 +1,309 @@
+                out.println(
+                        "Error: Activity not started, you do not "
+                                + "have permission to access it.");
+                break;
+            case ActivityManager.START_NOT_VOICE_COMPATIBLE:
+                out.println(
+                        "Error: Activity not started, voice control not allowed for: "
+                                + intent);
+                break;
+            case ActivityManager.START_NOT_CURRENT_USER_ACTIVITY:
+                out.println(
+                        "Error: Not allowed to start background user activity"
+                                + " that shouldn't be displayed for all users.");
+                break;
+            default:
+                out.println(
+                        "Error: Activity not started, unknown error code " + res);
+                break;
+        }
+
+        System.err.println("Activity is" + (launched?"":" not") + " started");
+        PendingIntent p;
+
+        return launched;
+    }
+
+    private void startXwayland(String[] args) throws IOException {
+        System.err.println("Starting Xwayland");
+        ExecHelper.exec(XwaylandPath, args);
+    }
+
+    class Service extends ITermuxX11Internal.Stub {
+        @Override
+        public boolean onTransact(int code, android.os.Parcel data, android.os.Parcel reply, int flags) throws RemoteException {
+            Starter.this.onTransact(code, data, reply, flags);
+            return super.onTransact(code, data, reply, flags);
+        }
+
+        @Override
+        public ParcelFileDescriptor getWaylandFD() throws RemoteException {
+            System.err.println("Got getWaylandFD");
+            try {
+                return Starter.this.getWaylandFD();
+            } catch (Throwable e) {
+                throw new RemoteException(e.getMessage());
+            }
+        }
+
+        @Override
+        public ParcelFileDescriptor getLogFD() throws RemoteException {
+            System.err.println("Got getLogFD");
+            System.err.println(logFD);
+            return logFD;
+        }
+
+        @Override
+        public void finish() throws RemoteException {
+            System.err.println("Got finish request");
+            handler.postDelayed(Starter.this::onFinish, 10);
+        }
+    }
+
+    private void exit(int status) {
+        exit(50, status);
+    }
+    private void exit(int delay, int status) {
+        handler.postDelayed(() -> System.exit(status), delay);
+    }
+
+    private native void checkXdgRuntimeDir();
+    private native int createWaylandSocket();
+    private native boolean checkWaylandSocket();
+    private native int openLogFD();
+    @SuppressWarnings("FieldMayBeFinal")
+    private static Handler handler;
+    static {
+        Looper.prepare();
+        handler = new Handler();
+
+        @SuppressLint("SdCardPath")
+        final String libPath = "/data/data/com.termux/files/usr/libexec/termux-x11/libstarter.so";
```

Comment:
Considering adding support for 3rd party apps, I won't considering hardcoding this path...

I would prefer using this only if `DEFAULT_PREFIX` is unset.. just like in `starter.c`

Reply:
Starter is needed only to trigger a termux-x11 to start without `Context`. 3rd party apps will be able to trigger termux-x11 inside `Activity`. I will add notes about that ot Readme before release. Or at least after reviewing and before merging this pull request.